src/middlewared/middlewared/plugins/init_shutdown_script.py

```diff
@@ -109,13 +109,26 @@ class Config:
         Str('when')
     )
     def execute_init_tasks(self, when):
-        preinit_tasks = self.middleware.call_sync(
+        if when == 'POSTINIT':
+            # Sentinel file to tell we have gone far enough in the boot process.
```

Comment:
I feel both this tasks (creating `/tmp/.bootready` and sending ready event) should be moved to some private method like `core.notify_postinit` and it be called from `ix-postinit` script, do you think that can be sane? `initshutdownscript` must only do user-defined scripts.

Reply:
I agree, this approach would indeed make more sense. Bringing up the changes asap.